backend/src/test/java/com/qwarty/auth/service/AuthServiceTest.java

```diff
@@ -0,0 +1,175 @@
+package com.qwarty.auth.service;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+
+import com.qwarty.auth.dto.LoginAuthRequestDTO;
+import com.qwarty.auth.dto.LoginAuthResponseDTO;
+import com.qwarty.auth.dto.SignupAuthRequestDTO;
+import com.qwarty.auth.model.User;
+import com.qwarty.auth.repository.RefreshTokenRepository;
+import com.qwarty.auth.repository.UserRepository;
+import com.qwarty.exception.CustomException;
+import com.qwarty.exception.CustomExceptionCode;
+import jakarta.servlet.http.HttpServletResponse;
+import java.util.Date;
+import java.util.Optional;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@ExtendWith(MockitoExtension.class)
+class AuthServiceTest {
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private RefreshTokenRepository refreshTokenRepository;
+
+    @Mock
+    private JwtService jwtService;
+
+    @Mock
+    private PasswordEncoder passwordEncoder;
+
+    @Mock
+    private AuthenticationManager authenticationManager;
+
+    @InjectMocks
+    private AuthService authService;
+
+    @Test
+    void signup_successfulRegistration() {
+        SignupAuthRequestDTO request = new SignupAuthRequestDTO("newuser", "new@example.com", "password123");
```

Comment:
```suggestion
        String username = "newuser";
        String email = "new@example.com";
        String password = "password123";
        String encodedPassword = "encodedPassword";
        SignupAuthRequestDTO request = new SignupAuthRequestDTO(username, email, password);
```

Can we extract the literals used here to ensure consistency throughout the whole test? Should help with readability instead since we're comparing and passing in descriptive variable names instead of string literals.

Same comment for other tests

Reply:
right, makes sense, will change these